src/ILCompiler/src/Program.cs

```diff
@@ -315,7 +316,7 @@ private int Run(string[] args)
                 }
                 else if (_nativeLib)
                 {
-                    EcmaModule module = (EcmaModule)typeSystemContext.SystemModule;
+                    EcmaModule module = typeSystemContext.GetModuleFromPath(typeSystemContext.InputFilePaths.First().Value);
```

Comment:
This does not look right. Could you please explain what it is trying to fix?

Reply:
cc @MichalStrehovsky

Reply:
This is to fix the multimodule build. After a bit of tinkering I noticed that in multimodule mode `ilc` ignores all methods that belong to the system module because of this the native startup method wasn't being compiled into the resulting object file (because I set the owning module to the system module). This change simply sets the owning module to the first input module (anyone is fine) so the startup method gets included in the object file

Reply:
Ok, make sense. It may be useful to add comment.

Reply:
It might also be possible to shuffle things around enough so that this is the `GeneratedAssembly` (this is a property on `CompilationModuleGroup`). It's not an `EcmaModule`, but it doesn't seem like it's strictly necessary for `NativeLibraryInitializerRootProvider` to take an `EcmaModule` (`ModuleDesc` should be fine).

Reply:
Agree - that would look cleaner.